src/schemathesis/cli/junitxml.py

```diff
@@ -0,0 +1,41 @@
+import platform
+from typing import List, Optional
+
+import attr
+from click.utils import LazyFile
+from junit_xml import TestCase, TestSuite, to_xml_report_file
+
+from ..models import Status
+from ..runner import events
+from .handlers import EventHandler, ExecutionContext, get_unique_failures
+
+
+@attr.s(slots=True)  # pragma: no mutate
+class JunitXMLHandler(EventHandler):
+    file_handle: LazyFile = attr.ib()  # pragma: no mutate
+    test_cases: List = attr.ib(factory=list)  # pragma: no mutate
+    start_time: Optional[float] = attr.ib(default=None)  # pragma: no mutate
+
+    def handle_event(self, context: ExecutionContext, event: events.ExecutionEvent) -> None:
+        if isinstance(event, events.Initialized):
+            self.start_time = event.start_time
+        if isinstance(event, events.AfterExecution):
+            test_case = TestCase(f"{event.result.method} {event.result.path}")
+            test_case.elapsed_sec = event.elapsed_time
+            self.test_cases.append(test_case)
```

Comment:
Could you, please, append at the end of this branch to avoid list access at all?